### PR TITLE
CRM-18364: Reply-To field depopulates when saving then continuing a d…

### PR DESCRIPTION
**Pull request by jitendrapurohit**
…raft mailing https://issues.civicrm.org/jira/browse/CRM-18364

### CODE REVIEW

File: ang/crmMailing/FromAddress.js [hunk 1]
```diff
@@ -18,7 +18,10 @@
           var addr = crmFromAddresses.getByLabel(newValue);
           mailing.from_name = addr.author;
           mailing.from_email = addr.email;
-          mailing.replyto_email = addr.email;
+          // CRM-18364: set replyTo as from_email only if custom replyTo is disabled in mail settings.
+          if (!CRM.crmMailing.enableReplyTo) {
+            mailing.replyto_email = crmFromAddresses.getByAuthorEmail(mailing.from_name, mailing.from_email, true).label
```
Comment (JKingsnorth) [inline ang/crmMailing/FromAddress.js]: Missing ;   ?

Comment (seamuslee001) [inline ang/crmMailing/FromAddress.js]: Also my reading of the replyto column would suggest all we need is the email not the full label here.

**Comment by seamuslee001** [comment]
I have performed the following test. (note this is against 4.6 but code is the same)

1. copy a CiviMail that has previously been used - whilst custom replyto is set to off. 
2. Change the from address and verified that the changed from email address appears in the replyto_email and from_email columns of civicrm_mailing. (This satisfies CRM-17307)
3. Saved draft and then turned on Custom replyto. Set the reply to to be a different email address from the from email. Verified in the database that the correct replyto email was populated and that the fromEmail stayed the same. 
4. Went to the draft and unscheduled mailing window and clicked continue on this test mailing, Verified that the replyto field and the from email field loaded with the correct addresses in them.

File: ang/crmMailing/FromAddress.js (hunk 1 above)
Comment (jitendrapurohit) [inline ang/crmMailing/FromAddress.js]: @JKingsnorth fixed the syntax error. Thanks!

**Comment by jitendrapurohit** [comment]
@seamuslee001 Thanks for testing this with different use-cases. 

For the reading of replyTo column, as I found that replyTo column is being saved as full label when the custom replyTo is enabled(i.e, before this PR and #7408), I've kept the behavior as consistent when this column is disabled. This is also a need for the HTML [dropdown to set its value by default].(https://github.com/civicrm/civicrm-core/blob/master/ang/crmMailing/BlockMailing.html#L45)

**Comment by seamuslee001** [comment]
@jitendrapurohit Heya, happy to roll with it as written, the only thing is perhaps maybe the way the replyto has been set up in this ab testing stuff is different to how the column is set up as per the comment on the [xml schema](https://github.com/civicrm/civicrm-core/blob/master/xml/schema/Mailing/Mailing.xml#L157) It would seem to be that the database schema is expecting email only but we are populating with the full label. However given what you have pointed out happy for this to roll as is. Do you want to open a PR to backport to 4.6 or are you fine for me to?

**Comment by jitendrapurohit** [comment]
Back-ported the fix for 4.6 at https://github.com/civicrm/civicrm-core/pull/8137. Thanks:)

**Pull request merged**